renderdoc/driver/vulkan/vk_resources.cpp

```diff
@@ -3216,6 +3216,326 @@ VkImageAspectFlags FormatImageAspects(VkFormat fmt)
+
+          loadOpTable[remappedIndex] = ci.pAttachments[index].loadOp;
+
+          if(multiviewViewMaskTable)
+          {
+            multiviewViewMaskTable[remappedIndex] |= ci.pSubpasses[s].viewMask;
+          }
+        }
+        ++pResolveRunner;
+      }
+    }
+    if(pDepthStencilAttachment)
+    {
+      uint32_t index = pDepthStencilAttachment->attachment;
+
+      if(index < ci.attachmentCount)
+      {
+        VkAttachmentLoadOp depthStencilLoadOp = ci.pAttachments[index].loadOp;
+        VkAttachmentLoadOp stencilLoadOp = ci.pAttachments[index].stencilLoadOp;
+
+        uint32_t remappedIndex = indexRemapTable[index];
+        RDCASSERT(remappedIndex < arrayCount);
+
+        if(IsStencilFormat(ci.pAttachments[index].format))
+        {
+          // VK_KHR_separate_depth_stencil_layouts
+          VkAttachmentDescriptionStencilLayoutKHR *separateStencil =
+              (VkAttachmentDescriptionStencilLayoutKHR *)FindNextStruct(
+                  &ci.pAttachments[index],
+                  VK_STRUCTURE_TYPE_ATTACHMENT_DESCRIPTION_STENCIL_LAYOUT_KHR);
+
+          if(separateStencil)
+          {
+            loadOpTable[remappedIndex + 1] = stencilLoadOp;
+          }
+          else
+          {
+            // make depthstencil VK_ATTACHMENT_LOAD_OP_LOAD if either depth or stencil is
+            // VK_ATTACHMENT_LOAD_OP_LOAD
+            if(depthStencilLoadOp != VK_ATTACHMENT_LOAD_OP_LOAD)
+            {
+              depthStencilLoadOp = stencilLoadOp;
+            }
+          }
+        }
+
+        loadOpTable[remappedIndex] = depthStencilLoadOp;
+
+        if(multiviewViewMaskTable)
+        {
+          multiviewViewMaskTable[remappedIndex] |= ci.pSubpasses[s].viewMask;
+        }
+      }
+    }
+  }
+}
+
+RenderPassInfo::~RenderPassInfo()
+{
+  delete[] imageAttachments;
+  delete[] loadOpTable;
+  delete[] multiviewViewMaskTable;
+}
+
+FramebufferInfo::FramebufferInfo(const VkFramebufferCreateInfo &ci)
+{
+  // *2 in case we need separate barriers for depth and stencil, +1 for the terminating null
+  // attachment info
+  uint32_t arrayCount = ci.attachmentCount * 2 + 1;
+
+  imageAttachments = new AttachmentInfo[arrayCount];
+  RDCEraseMem(imageAttachments, arrayCount * sizeof(imageAttachments[0]));
+
+  width = ci.width;
+  height = ci.height;
+  layers = ci.layers;
+}
+FramebufferInfo::~FramebufferInfo()
+{
+  delete[] imageAttachments;
+}
+
+bool FramebufferInfo::AttachmentFullyReferenced(size_t attachmentIndex, const RenderPassInfo *rpi)
+{
+  VkResourceRecord *att = imageAttachments[attachmentIndex].record;
+  // if framebuffer doesn't reference the entire image
+  if(att->resInfo->imageInfo.extent.width != width || att->resInfo->imageInfo.extent.height != height)
+  {
+    return false;
+  }
+  // if view doesn't reference the entire image
+  if(att->viewRange.baseArrayLayer != 0 ||
+     att->viewRange.layerCount() != (uint32_t)att->resInfo->imageInfo.layerCount)
```

Comment:
One final case to worry about - this should also check the base mip level and mip count.

Reply:
What does it mean for a framebuffer with attachments referencing only parts of the mip chain?

Reply:
It means that we're not completely writing to the whole image, only a part of it. Another part of the image could still be read similar to the case with layers so we shouldn't mark it as completely written. I don't think from the top of my head that it's possible for a framebuffer to reference multiple mips so this is probably the case with any image that has mips, but it's easier just to replicate the same check I think.

Reply:
Ok, I understand now. Thank you for pointing that out!